controller/k8s/api_test.go

```diff
@@ -974,19 +974,19 @@ metadata:
 		},
 	} {
 		tt := tt // pin
-		for _, enableInformers := range []bool{
+		for _, retry := range []bool{
 			false,
 			true,
 		} {
-			enableInformers := enableInformers // pin
-			t.Run(fmt.Sprintf("%d/enableInformers:%t", i, enableInformers), func(t *testing.T) {
-				api, objs, err := newAPI(enableInformers, []string{tt.podConfig}, tt.extraConfigs...)
+			retry := retry // pin
```

Comment:
Just if you have time, could you explain what the pinning does? Thank you.

Reply:
The call below (`t.Run(fmt.Sprintf("%d/retry:%t", i, retry), func(t *testing.T)`...) uses `retry` inside the closure in that second argument. If we don't "pin" that variable, when those two closures run it's possible they will use the current value of `retry`, which in this case would be the last value of the array, `true`. If we "pin" it, then each closure will instead grab a copy of that value, a different copy for each iteration.
I've created a simpler example [here](https://play.golang.org/p/W0imMglYA-U).

Reply:
So I got it right and it's to avoid race conditions but just the name of the var within the closure visibility scope is the same as the name of the var created outside of the visibility scope. And Golang allows to define both of them together.

Reply:
Actually this is amazing comparing to other languages even though it looks tricky first. Thank you for the explanation.